### PR TITLE
✨ listen to wallet events in payment system

### PR DESCRIPTION
**Pull request by matusdrobuliak66**
## What do these changes do?
✨ listen to wallet events in the payment system

There will be 3 PRs:
- [x] listen to wallet events
- [ ] move auto recharge functionality from the webserver to the payment service
- [ ] make it idempotent (in case of payment service received multiple messages)

## How to test
new test `services/payments/tests/unit/test_services_auto_recharge_listener.py`

## Dev Checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by pcrespov — APPROVED** [review]

File: services/payments/src/simcore_service_payments/services/auto_recharge_listener.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        )
+        return subscribed_queue
+
+
+def setup_auto_recharge_listener(app: FastAPI) -> None:
+    async def _on_startup() -> None:
+        app.state.auto_recharge_rabbitmq_consumer = await _subscribe_to_rabbitmq(app)
+
+    async def _on_shutdown() -> None:
+        # NOTE: We want to have persistent queue, therefore we will not unsubscribe
```
Comment (sanderegg) [inline services/payments/src/simcore_service_payments/services/auto_recharge_listener.py:32]: so you want to keep the queue alright. but what about the consumer? you should remove it and not just rely on rabbitmq cleaning up your client here.

**Review by sanderegg — COMMENTED** [review]
I agree that you do not want to delete the queue. but nevertheless the consumer of the queue should be deleted

File: packages/service-library/tests/rabbitmq/test_rabbitmq.py [hunk 2]
```diff
@@ -435,3 +435,20 @@ async def test_rabbit_not_using_the_same_exchange_type_raises(
     # now do a second subscribtion wiht topics, will create a TOPICS exchange
     with pytest.raises(aio_pika.exceptions.ChannelPreconditionFailed):
         await client.subscribe(exchange_name, mocked_message_parser, topics=[])
+
+
+@pytest.mark.no_cleanup_check_rabbitmq_server_has_no_errors()
```
Comment (sanderegg) [inline packages/service-library/tests/rabbitmq/test_rabbitmq.py:440]: that is an interesting an fun decorator... why is that?

**Review by sanderegg — APPROVED** [review]
I'd like to understand the decorator. maybe you can tell me tomorrow.

**Review by GitHK — COMMENTED** [review]
I see that the PR is not fully ready

File: packages/service-library/tests/rabbitmq/test_rabbitmq.py (hunk 2 above)
Comment (matusdrobuliak66) [reply packages/service-library/tests/rabbitmq/test_rabbitmq.py:440]: Yes, we can check together

**Pull request merged**